endpoints/openrtb2/auction.go

```diff
@@ -1641,6 +1664,18 @@ func (deps *endpointDeps) validateApp(req *openrtb_ext.RequestWrapper) error {
 	return err
 }
 
+func (deps *endpointDeps) validateDOOH(req *openrtb_ext.RequestWrapper) error {
+	if req.DOOH == nil {
+		return nil
+	}
+
+	if req.DOOH.ID == "" && len(req.DOOH.VenueType) == 0 {
+		return errors.New("request.dooh should include at least one of request.dooh.id or request.dooh.venuetype.")
+	}
+
+	return nil
+}
```

Comment:
I don't think we should add this validation. From the OpenRTB spec:

> At a minimum, it is useful to provide id and/or venuetypeid, but this is not strictly required.

If it's not strictly required, I don't think we want to reject requests which do not supply that info.

Reply:
@SyntaxNode How strongly do you feel about this  ?

I opted to do something similar to the basic validation done for "site" objects - that is, despite the openrtb spec stating:
> 3.2.13 - Object: Site
> ...  At a minimum, it is useful to provide a site ID or page URL, but this is not strictly required.

There's basic validation for either of them [here](https://github.com/prebid/prebid-server/blob/2c2adcf9e69cb38cd2494adfecc1c330e4a4f579/endpoints/openrtb2/auction.go#L1671-L1673)

I assume the rationale is that without either, someone's attempting to transact near-entirely-anonymous inventory which is.... dubious... at best.

Reply:
Hi @minaguib, we discussed as a team and would like to confirm the following:
Bidders will receive inventory identifiers in `req.imp[].ext.prebid.bidder.{bidder}`. Are `dooh.id` and `dooh.venueType` global or bidder specific? How will they be used in future? Should the entire request be rejected if these parameters are not specified?
We are not sure if in this case we need to follow the existing pattern with `site.id` and `site.page` validation. In any case, for these site parameters, one could argue that these parameters aren't actually required. The validation rule predates me but something like `site.id` is often defined in the bidder parameters as well and suffices in the absence of the global value.

Reply:
FWIW - I would lean towards trusting the DOOH committee to determine which fields should be required in Prebid Server.

Reply:
Opted to keep the check. If through adoption we hear back it's too strict, we can relax it.  The opposite (introducing stricter validation without breakage) would be harder.